### PR TITLE
hybrid-mode: Only remove binding when default command used

### PR DESCRIPTION
**Pull request by justbur**
This one's for you @mkcode  ;-)

https://github.com/syl20bnr/spacemacs/issues/1244#issuecomment-130760355

### CODE REVIEW

**Comment by mkcode** [comment]
:heart:

**Comment by justbur** [comment]
@mkcode One more for you, which I think will resolve any issues you might have. I added a keymap, `hybrid-mode-insert-state-map`, which I only activate in insert state when hybrid mode is active.

**Comment by justbur** [comment]
This now fixes the issue raised in #2650, but I think in a little cleaner way.

File: spacemacs/local/hybrid-mode/hybrid-mode.el [hunk 1]
```diff
@@ -56,27 +93,31 @@
...
-  (define-key evil-insert-state-map
-    (read-kbd-macro evil-toggle-key) nil)
-  ;; see if the above works withou this
-  ;; (when (bound-and-true-p evil-escape-mode)
-  ;;   (define-key evil-insert-state-map "f" 'evil-escape-insert-state))
-  )
+        hybrid-mode-insert-state-cursor-backup evil-insert-state-cursor
+        evil-insert-state-cursor hybrid-mode-insert-state-cursor)
+  (copy-face 'hybrid-mode-insert-face 'spacemacs-insert-face)
+  (dolist (key-bnd hybrid-mode-default-evil-insert-state-bindings)
```
Comment (syl20bnr) [inline spacemacs/local/hybrid-mode/hybrid-mode.el]: Can you add a comment to explain what is done by the dolist ? Although it is easy to understand the "what is done" it is not so obvious to deduce the "why" which is to remove _non customized keys_ from the insert state map.

**Comment by syl20bnr** [comment]
@justbur I merged #2650, you'll need to rebase ;-)

**Comment by justbur** [comment]
@syl20bnr  I think this is ready. I put all of the spacemacs configuration in the spacemacs layer, too. It's all squashed into one commit

**Comment by syl20bnr** [comment]
Very nice :100: 
Thank you !
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**